plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/core.py

```diff
@@ -0,0 +1,218 @@
+    version: int
+       the version of the commandline input parsing. The parsing will probably evolve in the near
+       future and several versions may temporarily coexist.
+       s
+    """
+
+    def __init__(
+        self,
+        optimizer: str,
+        budget: int,
+        num_workers: int,
+        noisy: bool,
+        maximize: bool,
+        seed: Optional[int],
+        version: int,
+    ):
+        self.config: Optional[DictConfig] = None
+        self.launcher: Optional[Launcher] = None
+        assert version == 1, "Only version 1 of API is currently available"
+        self.job_results = None
+        self.optimizer = optimizer
+        self.noisy = noisy
+        self.budget = budget
+        self.num_workers = num_workers
+        self.seed = seed
+        self._direction = -1 if maximize else 1
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.launcher = Plugins.instantiate_launcher(
+            config=config, config_loader=config_loader, task_function=task_function
+        )
+
+    def sweep(self, arguments: List[str]) -> Any:
+        assert self.config is not None
+        assert self.launcher is not None
+        # Construct the parametrization
+        params: Dict[str, Union[str, int, float, ng.p.Parameter]] = {}
+        for s in arguments:
+            key, value = s.split("=", 1)
+            params[key] = make_parameter(value)
+        parametrization = ng.p.Dict(**params)
+        parametrization.descriptors.deterministic_function = not self.noisy
+        parametrization.random_state.seed(self.seed)
+        # log and build the optimizer
+        name = "maximization" if self._direction == -1 else "minimization"
+        log.info(
+            "NevergradSweeper(optimizer=%s, budget=%s, num_workers=%s) %s",
+            self.optimizer,
+            self.budget,
+            self.num_workers,
+            name,
+        )
+        log.info("with parametrization %s", parametrization)
+        log.info("Sweep output dir : %s", self.config.hydra.sweep.dir)
+        optimizer = ng.optimizers.registry[self.optimizer](
+            parametrization, self.budget, self.num_workers
+        )
+        # loop!
+        remaining_budget = self.budget
+        all_returns: List[Any] = []
+        while remaining_budget > 0:
+            batch = min(self.num_workers, remaining_budget)
+            remaining_budget -= batch
+            candidates = [optimizer.ask() for _ in range(batch)]
+            overrides = list(
+                tuple(f"{x}={y}" for x, y in c.value.items()) for c in candidates
+            )
+            returns = self.launcher.launch(overrides)
+            # would have been nice to avoid waiting for all jobs to finish
+            # aka batch size Vs steady state (launching a new job whenever one is done)
```

Comment:
This is something I will consider in a future version.
For now launchers are launching in batches.
As of Hydra 1.0 (which is not yet released, Hydra requires Python 3.6 which makes an async launcher API more plausible).